base/strings/basic.jl

```diff
@@ -10,24 +10,44 @@ next(s::AbstractString, i::Integer) = next(s,Int(i))
 string() = ""
 string(s::AbstractString) = s
 
-bytestring() = ""
-bytestring(s::Vector{UInt8}) =
+"""
+    String(s::AbstractString)
+
+Convert a string to a contiguous byte array representation encoded as UTF-8 bytes.
+This representation is often appropriate for passing strings to C.
+"""
+String(s::AbstractString) = print_to_string(s)
+String(s::String) = s
+
+"""
+    String(v::Vector{UInt8})
+
+Wrap a vector of bytes encoding string data as UTF-8 in a `String` object.
+The resulting `String` object takes ownership of the array.
+"""
+String(s::Vector{UInt8}) =
```

Comment:
That's because one of the calls to `String(bytes)` in datafmt.jl should have been `String(copy(bytes))` ... I have a patch forthcoming.